packages/stryker-vue-mutator/src/VueMutator.ts

```diff
@@ -0,0 +1,80 @@
+import { Config } from 'stryker-api/config';
+import { File } from 'stryker-api/core';
+import { Mutant, Mutator, MutatorFactory } from 'stryker-api/mutant';
+const compiler = require('vue-template-compiler');
+
+export default class VueMutator implements Mutator {
+  private mutators: { [name: string]: Mutator; };
+
+  constructor(private config: Config) { }
+
+  mutate(inputFiles: File[]): Mutant[] {
+    this.initializeMutators();
+    let mutants: Mutant[] = [];
+
+    inputFiles.forEach(file => {
+      if (file.name.endsWith('.vue')) {
+        const script = compiler.parseComponent(file.textContent).script;
+        const mutator = this.getVueScriptMutator(script);
+        const vueFile = new File(
+          file.name + '.js', /* TODO: Remove this once all mutators understand that vue files are OK to mutate */
+          file.textContent.substring(script.start, script.end)
+        );
+        const vueMutants = mutator.mutate([vueFile]);
+        vueMutants.forEach(mutant => {
+          mutant.fileName = file.name; /* TODO: Remove this once all mutators understand that vue files are OK to mutate */
+          mutant.range[0] += script.start;
+          mutant.range[1] += script.start;
+        });
+        mutants = mutants.concat(vueMutants);
+      } else {
+        const mutator = this.getMutator(file);
+        mutants = mutants.concat(mutator.mutate([file]));
+      }
+    });
+
+    return mutants;
+  }
+
+  private initializeMutators() {
```

Comment:
Can we move this method to a separate file? That way the unit tests can be independent of the current `MutatorFactory` way of working. We will remove the mutator factory as part of #667  .

Reply:
Fixed